packages/mux-player-react/src/index.tsx

```diff
@@ -0,0 +1,186 @@
+  onEnded?: EventListener;
+  onError?: EventListener;
+  onPlayerReady?: EventListener;
+} & Partial<MuxMediaPropTypes> &
+  Partial<VideoApiAttributes>;
+
+const MuxPlayerInternal = React.forwardRef<
+  MuxPlayerRefAttributes,
+  MuxPlayerProps
+>(({ children, ...props }, ref) => {
+  return React.createElement(
+    "mux-player",
+    toNativeProps({ ...props, ref }),
+    children
+  );
+});
+
+const useEventCallbackEffect = (
+  type: string,
+  ref: // | ((instance: EventTarget | null) => void)
+  React.MutableRefObject<EventTarget | null> | null | undefined,
+  callback: EventListener | undefined
+) => {
+  return useEffect(() => {
+    const eventTarget = ref?.current;
+    if (!eventTarget || !callback) return;
+    eventTarget.addEventListener(type, callback);
+    return () => {
+      eventTarget.removeEventListener(type, callback);
+    };
+  }, [ref?.current, callback]);
+};
+
+const usePlayer = (
+  ref: // | ((instance: EventTarget | null) => void)
+  React.MutableRefObject<MuxPlayerElement | null> | null | undefined,
+  props: MuxPlayerProps
+) => {
+  const {
+    onLoadStart,
+    onLoadedMetadata,
+    onProgress,
+    onDurationChange,
+    onVolumeChange,
+    onRateChange,
+    onResize,
+    onWaiting,
+    onPlay,
+    onPlaying,
+    onTimeUpdate,
+    onPause,
+    onSeeking,
+    onSeeked,
+    onEnded,
+    onError,
+    onPlayerReady,
+    metadata,
+    paused,
+    ...remainingProps
+  } = props;
+  useObjectPropEffect("metadata", metadata, ref);
+  useObjectPropEffect(
+    "paused",
+    paused,
+    ref,
+    (playerEl: HTMLMediaElement, pausedVal: boolean) => {
+      if (pausedVal) {
+        playerEl.pause();
+      } else {
+        playerEl.play();
+      }
+    }
+  );
+  useEventCallbackEffect("loadstart", ref, onLoadStart);
+  useEventCallbackEffect("loadedmetadata", ref, onLoadedMetadata);
+  useEventCallbackEffect("progress", ref, onProgress);
+  useEventCallbackEffect("durationchange", ref, onDurationChange);
+  useEventCallbackEffect("volumechange", ref, onVolumeChange);
+  useEventCallbackEffect("ratechange", ref, onRateChange);
+  useEventCallbackEffect("resize", ref, onResize);
+  useEventCallbackEffect("waiting", ref, onWaiting);
+  useEventCallbackEffect("play", ref, onPlay);
+  useEventCallbackEffect("playing", ref, onPlaying);
+  useEventCallbackEffect("timeupdate", ref, onTimeUpdate);
+  useEventCallbackEffect("pause", ref, onPause);
+  useEventCallbackEffect("seeking", ref, onSeeking);
+  useEventCallbackEffect("seeked", ref, onSeeked);
+  useEventCallbackEffect("ended", ref, onEnded);
+  useEventCallbackEffect("error", ref, onError);
+  useEventCallbackEffect("playerready", ref, onPlayerReady);
+  return [remainingProps];
+};
+
+const MuxPlayer = React.forwardRef<MuxPlayerRefAttributes, MuxPlayerProps>(
+  (props, ref) => {
+    const {
+      /** @TODO Remove these once defaults are added to mux-player (CJP) */
+      forwardSeekOffset = 10,
+      backwardSeekOffset = 10,
```

Comment:
I'll add these asap. yes, can we go with 10s instead of 30s? that feels a bit too much for most videos

Reply:
completely wild thought: have these as 10s for videos shorter than 5 minutes and 30s for videos longer than 5 minutes?

Reply:
that would have to be a followup since we'll render the icons before we know the media duration, but cool thought